src/merkle_tree/src/merkle_tree.cairo

```diff
@@ -35,30 +35,28 @@ trait HasherTrait<T> {
 // Hasher representations.
 
 #[derive(Drop, Copy)]
-struct Hasher {}
+pub struct Hasher {}
 
 /// Hasher impls.
 
-mod pedersen {
-    use hash::HashStateTrait;
+pub mod pedersen {
     use super::{Hasher, HasherTrait};
 
-    impl PedersenHasherImpl of HasherTrait<Hasher> {
+    pub impl PedersenHasherImpl of HasherTrait<Hasher> {
```

Comment:
no i'm talking about the trait itself not the pub

Reply:
I guess so that if you init it from pedersen it'll use pedersen hashing, and if using poseidon, it'll be using poseidon stuff
Idk, I'm just updating what is already there ^^'
If this needs some re-work (which it does) I'd open an issue